spec/controllers/chapters_controller_spec.rb

```diff
@@ -361,6 +361,33 @@
         it_redirects_to_with_error(work_path(work), "Sorry, you don't have permission to access the page you were trying to reach.")
       end
     end
+
+    context "when co-creator is logged in" do
+      let(:co_creator) { create(:user) }
+      let!(:co_second_chapter) { create(:chapter, work: work, posted: true, authors: [user.pseuds.first, co_creator.pseuds.first]) }
+      let!(:co_third_chapter) { create(:chapter, work: work, posted: true, authors: [user.pseuds.first, co_creator.pseuds.first]) }
+
+      before do
+        fake_login_known_user(co_creator)
+      end
+
+      it "removes user, gives notice, and redirects to work when user removes themselves" do
+        get :edit, params: { work_id: work.id, id: co_second_chapter.id, remove: "me" }
+        expect(assigns[:chapter].pseuds).to eq [user.pseuds.first]
+        it_redirects_to_with_notice(work_path(work), "You have been removed as a creator from the chapter.")
+      end
+
+      it "delegates to work removal of user after user removed from last co-created chapter" do
+        get :edit, params: { work_id: work.id, id: co_second_chapter.id, remove: "me" }
+        expect(assigns[:work].pseuds).to eq [user.pseuds.first, co_creator.pseuds.first]
+        get :edit, params: { work_id: work.id, id: co_third_chapter.id, remove: "me" }
+
+        expect(co_second_chapter.reload.pseuds).to eq [user.pseuds.first]
+        expect(co_third_chapter.reload.pseuds).to eq [user.pseuds.first]
+
+        it_redirects_to_with_notice(edit_work_path(work, remove: "me"), "You have been removed as a creator from the chapter.")
```

Comment:
This check looks odd: reading the code, we should be redirected without any notice. Looks like the notice is left over from the first `get :edit`.

https://stackoverflow.com/questions/7850650/making-two-requests-to-the-same-controller-in-rails-integrations-specs

So I think we need to keep to one `get` per spec and do all the setup in `before` blocks:
- Set up 2 chapters.
- In context 1, no extra setup, test removing co-creator from last chapter, removing them from the work.
- In context 2, add a third co-created chapter, test removing co-creator from just one chapter, leaving them on the work.